### PR TITLE
 WT-7408 API to return row and byte counts for objects and cursor ranges

### PR DESCRIPTION
**Pull request by keithbostic**
Re-open pull request for WT-7408.

### CODE REVIEW

**Comment by ddanderson** [comment]
@keithbostic, I just pushed changes to fix the SWIG/Python support, and modified the test case slightly.  In theory, there's an easier way to do the SWIG stuff, you're supposed to be able to match on two side by side uint64_t * arguments, but I couldn't get that to work.  So the solution appears to be a little more complex.

File: test/format/checkpoint.c [hunk 1]
```diff
@@ -73,12 +73,17 @@ checkpoint(void *arg)
     conn = g.wts_conn;
     testutil_check(conn->open_session(conn, NULL, NULL, &session));
 
-    for (secs = mmrand(NULL, 1, 10); !g.workers_finished;) {
-        if (secs > 0) {
+    /*
```
Comment (quchenhao) [inline test/format/checkpoint.c:76]: You intend to write something here?

File: src/session/session_api.c [hunk 2]
```diff
@@ -1238,6 +1238,256 @@ __session_join(
...
+static int
+__session_range_cursor_table(
+  WT_CURSOR_TABLE *start, WT_CURSOR_TABLE *stop, uint64_t *row_countp, uint64_t *byte_countp)
+{
+    WT_DECL_RET;
+    uint64_t row_count;
+    u_int i;
+
+    /* Set the number of rows from one column group, sum the column group byte counts. */
+    row_count = 0; /* [-Wconditional-uninitialized] */
```
Comment (quchenhao) [inline src/session/session_api.c:1301]: It feels a bit odd to me that we have to count it multiple times. Should we return the min or max here?

File: src/btree/bt_range_stat.c [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+                break;
+            }
+        else
+            ++missing_addr;
+        WT_REF_UNLOCK(ref, previous_state);
+
+        /*
+         * An adjustment to improve accuracy: assume the key takes up half of the range in the slot
+         * itself, on the first and last slots. This also makes the case where the two cursors are
+         * on the same leaf page work return something reasonable, a range that isn't insane. If we
```
Comment (quchenhao) [inline src/btree/bt_range_stat.c:221]: typo work

**Comment by quchenhao** [comment]
I had a browse of the PR. I got the general idea of what it's doing but don't understand the details, especially the code about address cookies. We can have a chat later.

File: test/format/checkpoint.c (hunk 1 above)
Comment (keithbostic) [reply test/format/checkpoint.c]: Thanks, fixed in 0854f96b9.

File: src/btree/bt_range_stat.c (hunk 3 above)
Comment (keithbostic) [reply src/btree/bt_range_stat.c]: Thanks, fixed in 0854f96b9.

File: src/session/session_api.c (hunk 2 above)
Comment (keithbostic) [reply src/session/session_api.c:1301]: I'm assuming all of the column groups have the same number of rows (because they each have some subset of the columns), so we only count that once, but the total number of bytes in all of the column groups is interesting, and so I'm aggregating that. Does that make sense to you?

**Comment by keithbostic** [comment]
> I had a browse of the PR. I got the general idea of what it's doing but don't understand the details, especially the code about address cookies. We can have a chat later.

Thank you, @quchenhao, I've updated the PR based on your comments.

I should have noted this in the PR: as far as I'm concerned, this work is complete, or, at least as complete as it can be without a full review and/or better understanding of what the final product will look like. @agorrod and @keitharnoldsmith are holding off on that decision: once we get out of the current set of data corruption issues, Alex is planning to review the project and the potential issues around diverging from the 5.0 format, and once he's done that we'll start making decisions about where to go from here.

In summary, we're holding off on full review until Alex gets back to us.

**Comment by keithbostic** [comment]
There's no reason to pull this branch forward, discard for now.
